src/hooks/useWaitForNavigation.ts

```diff
@@ -8,21 +8,18 @@ type UseWaitForNavigation = (navigate: () => void) => () => Promise<void>;
  * Only use when navigating by react-navigation
  */
 export default function useWaitForNavigation(): UseWaitForNavigation {
-    const navigation = useNavigation();
     const resolvePromises = useRef<Array<() => void>>([]);
 
-    useEffect(() => {
-        const unsubscribeBlur = navigation.addListener('blur', () => {
-            resolvePromises.current.forEach((resolve) => {
-                resolve();
-            });
-            resolvePromises.current = [];
-        });
-
-        return () => {
-            unsubscribeBlur();
-        };
-    }, [navigation]);
+    useFocusEffect(
+        useCallback(() => {
```

Comment:
Was it necessary to use `useCallback` here with an empty dependency array? It seems unnecessary since the callback won't change.

Reply:
Yes, it is necessary. Otherwise, the callback will be run on every render. Quote from the [docs](https://reactnavigation.org/docs/use-focus-effect/?config=static):

> The effect will run whenever the dependencies passed to React.useCallback change, i.e. it'll run on initial render (if the screen is focused) as well as on subsequent renders if the dependencies have changed. If you don't wrap your effect in React.useCallback, the effect will run every render if the screen is focused.